File: LeetCode/Java Solutions/Sum of All Subset XOR Totals/sum.java
class Solution {
    public int subsetXORSum(int[] nums) {
        return backtrack(nums, 0, 0);
    }
    
    private int backtrack(int[] nums, int index, int currentXOR) {
        // If we've considered all elements, add the current XOR to the total sum
        if (index == nums.length) {
            return currentXOR;
        }
        // Recursively calculate the XOR sum including the current element
        int include = backtrack(nums, index + 1, currentXOR ^ nums[index]);
        // Recursively calculate the XOR sum excluding the current element
        int exclude = backtrack(nums, index + 1, currentXOR);
        // Return the sum of both choices
        return include + exclude;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.subsetXORSum(new int[]{1, 3}));  // Output: 6
        System.out.println(sol.subsetXORSum(new int[]{5, 1, 6}));  // Output: 28
        System.out.println(sol.subsetXORSum(new int[]{3, 4, 5, 6, 7, 8}));  // Output: 480
    }
}

File: LeetCode/Python Solutions/Sum of All Subset XOR Totals/sum.py
from typing import List

class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        def backtrack(index, current_xor):
            # If we've considered all elements, add the current XOR to the total sum
            if index == len(nums):
                return current_xor
            # Recursively calculate the XOR sum including the current element
            include = backtrack(index + 1, current_xor ^ nums[index])
            # Recursively calculate the XOR sum excluding the current element
            exclude = backtrack(index + 1, current_xor)
            # Return the sum of both choices
            return include + exclude
        
        # Start backtracking from index 0 with an initial XOR value of 0
        return backtrack(0, 0)

# Example usage
sol = Solution()
print(sol.subsetXORSum([1, 3]))  # Output: 6
print(sol.subsetXORSum([5, 1, 6]))  # Output: 28
print(sol.subsetXORSum([3, 4, 5, 6, 7, 8]))  # Output: 480
